src/Servers/Kestrel/Core/src/Internal/Http/MessageBody.cs

```diff
@@ -183,7 +202,11 @@ protected ValueTask<ReadResult> StartTimingReadAsync(ValueTask<ReadResult> readA
         {
             if (!readAwaitable.IsCompleted)
             {
-                TryProduceContinue();
+                ValueTask<FlushResult> continueTask = TryProduceContinueAsync();
```

Comment:
Shouldn't we indicate that the ValueTask result has been consumed? @stephentoub submitted a PR to do that for historical code: https://github.com/dotnet/aspnetcore/pull/31221

```cs
if (!continueTask.IsCompletedSuccessfully)
{
    return StartTimingReadAwaited(continueTask, readAwaitable, cancellationToken);
}
else
{
    continueTask.GetAwaiter().GetResult();
}
```

Reply:
Hmm.  I'd have expected the code in the PR to trigger CA2012, providing the same feedback as James.  Were no warnings issued for this?

Reply:
Your .editconfig changes are still there: https://github.com/dotnet/aspnetcore/blob/main/.editorconfig

I'm not familiar with configuring analyzers in editconfigs or CA2021. I'm not sure why it wasn't prompted.